lang/py/avro/errors.py

```diff
@@ -44,9 +44,9 @@ class IgnoredLogicalType(AvroWarning):
 
 class AvroTypeException(AvroException):
     """Raised when datum is not an example of schema."""
-    def __init__(self, expected_schema, datum):
+    def __init__(self, expected_schema, name, datum):
```

Comment:
These exceptions predate me, but I think it's a mistake that they have required arguments to their `__init__` methods. I'd suggest that they should be polymorphic in that their first argument should be either the concrete exception message _or_ these kwargs.

Reply:
Changed in https://github.com/apache/avro/pull/1156/commits/5d34e2c380df85a0d43894f46c1ea01ee2305c1d